modules/renter/renter.go

```diff
@@ -123,36 +122,110 @@ func newRenter(cs modules.ConsensusSet, tpool modules.TransactionPool, hdb hostD
 	if tpool == nil {
 		return nil, errNilTpool
 	}
+	if hc == nil {
+		return nil, errNilContractor
+	}
 	if hdb == nil {
 		// Nil hdb currently allowed for testing purposes. :(
 		// return nil, errNilHdb
 	}
 
 	r := &Renter{
-		cs:             cs,
-		hostDB:         hdb,
-		hostContractor: hc,
-
+		newFiles: make(chan *file),
 		files:    make(map[string]*file),
 		tracking: make(map[string]trackedFile),
 
-		persistDir: persistDir,
-		mu:         sync.New(modules.SafeMutexDelay, 1),
+		workerPool: make(map[types.FileContractID]*worker),
+
+		cs:             cs,
+		hostDB:         hdb,
+		hostContractor: hc,
+		persistDir:     persistDir,
+		mu:             sync.New(modules.SafeMutexDelay, 1),
+		tg:             new(sync.ThreadGroup),
 	}
 	if err := r.initPersist(); err != nil {
 		return nil, err
 	}
 
+	// Spin up the workers for the work pool.
+	contracts := hc.Contracts()
+	for _, c := range contracts {
+		err := r.addWorker(c.ID)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	go r.threadedRepairLoop()
```

Comment:
seems like `threadedDownloadLoop` isn't being spawned anywhere. Weren't we planning to merge it with the repair loop anyway? I thought that was the point of having workers, so that they could be shared between uploads/downloads.